packages/DApp/src/components/NotificationsList.tsx

```diff
@@ -3,83 +3,54 @@ import React from 'react'
 import styled from 'styled-components'
 import { AnimationNotification, AnimationNotificationMobile } from '../constants/animation'
 import { useContracts } from '../hooks/useContracts'
-import { NotificationItem, NotificationItemPlain } from './NotificationItem'
+import { NotificationItem, NotificationInfoItem, NotificationErrorItem } from './NotificationItem'
+import { useWaku } from '../providers/waku/provider'
 
-interface Props {
-  type: 'votes' | 'featured'
-}
-
-export function NotificationsList({ type }: Props) {
+export function NotificationsList() {
   const { notifications } = useNotifications()
-  const { votingContract, featuredVotingContract } = useContracts()
-
-  const getParsedLog = (log: any, type: 'votes' | 'featured') => {
-    switch (type) {
-      case 'votes': {
-        return votingContract.interface.parseLog(log)
-      }
-      case 'featured': {
-        return featuredVotingContract.interface.parseLog(log)
-      }
-    }
-  }
-
-  const parseVoting = (parsedLog: any) => {
-    let text = ''
-    if (parsedLog.name === 'VotingRoomStarted') {
-      text = ' voting room started.'
-    }
-    if (parsedLog.name === 'VotingRoomFinalized') {
-      if (parsedLog.args.passed == true) {
-        if (parsedLog.args.voteType === 1) {
-          text = ' is now in the communities directory!'
-        }
-
-        if (parsedLog.args.voteType === 0) {
-          text = ' is now removed from communities directory!'
-        }
-      }
-    }
-
-    return text
-  }
-
-  const parseFeatured = (parsedLog: any) => {
-    let text = ''
-    if (parsedLog.name === 'VotingStarted') {
-      text = 'Featured voting started.'
-    }
-    if (parsedLog.name === 'VotingFinalized') {
-      text = 'Featured voting was finalized.'
-    }
-
-    return text
-  }
+  const { votingContract } = useContracts()
+  const { isLoading, isError, restart } = useWaku()
 
   return (
     <NotificationsWrapper>
+      {isLoading && <NotificationInfoItem text="Connecting to a Waku node." />}
+
+      {!isLoading && isError && <NotificationErrorItem text="Failed connect to a Waku node." action={restart} />}
+
       {notifications.map((notification) => {
         if ('receipt' in notification) {
           return notification.receipt.logs.map((log) => {
-            const parsedLog = getParsedLog(log, type)
+            if (log.address !== votingContract.address) {
+              return
+            }
+
+            // this needs to be updated so it takes into account also interface of featuredVotingContract
```

Comment:
Any idea how to update this? 🤔

Reply:
By handling the other log case – `log.address === featuredVotingContract.address`.

Btw, I didn't leave this comment. In this PR, I just reverted your changes in this file since you've told me it wasn't in the first place and had side effects too.

For the exception you've also mention, I added only

```
if (log.address !== votingContract.address) {
              return
            }
```

Reply:
I think some of your changes could be used for it after all, but I didn't focus on fixing that here.